### PR TITLE
PropertiesExpander init support property resolver

### PR DESCRIPTION
**Pull request by qxo**
so we an using expression (ie:${basedir}) in checkstyle.properties

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertiesExpander.java [hunk 1]
```diff
@@ -49,10 +55,30 @@ public PropertiesExpander(Properties properties) {
         values = new HashMap<>(properties.size());
         for (Enumeration<?> e = properties.propertyNames(); e.hasMoreElements();) {
             final String name = (String) e.nextElement();
-            values.put(name, properties.getProperty(name));
+            String val = properties.getProperty(name);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertiesExpander.java:58]: `val` -> `value`

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertiesExpander.java [hunk 2]
```diff
@@ -49,10 +55,30 @@ public PropertiesExpander(Properties properties) {
...
+            val = resolveValue(properties, name, val, new HashSet<>());
+            values.put(name, val);
         }
     }
 
+    private static String resolveValue(final Properties properties, final String name,
+            final String val, final Set<String> resolveKeys) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertiesExpander.java:65]: remove "final" from all arguments.

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertiesExpander.java [hunk 3]
```diff
@@ -49,10 +55,30 @@ public PropertiesExpander(Properties properties) {
...
+        String ret = val;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertiesExpander.java:66]: please use full names for both variables.

**Review by romani — CHANGES_REQUESTED** [review]
please create issue on this and explain what is problem now and what is suggested behavior.
Issue will be used in release notes to route users to read more.
Issue number should be used in commit message.

We also need to update documentation files in xdoc folder, 

items:

**Comment by romani** [comment]
@qxo , please reply

**Comment by qxo** [comment]
> @qxo , please reply

I just saw it.
ok, I'll check the review.

**Comment by nmancus1** [comment]
@qxo please share any thoughts at https://github.com/checkstyle/checkstyle/issues/9173

**Comment by romani** [comment]
this PR is abandoned ,
@nmancus1 , please open new PR with update of code for issue that you are created.
please do not forget about documentation update on this feature.

**Pull request closed without merging**